FreeTAKServer/controllers/DatabaseControllers/table_controllers.py

```diff
@@ -47,11 +52,13 @@ def query(self, session, query, columns):
     def update(self, session, query, column_value):
         DataPackages = session.query(self.table).filter(
             text(query)).all()  # self.query(session, query, [column for column, value in column_value.items()])
-        for dp in DataPackages:
-            for column, value in column_value.items():
-                setattr(dp, column, value)
-        session.commit()
-
+        if len(DataPackages) >0:
+            for dp in DataPackages:
+                for column, value in column_value.items():
+                    setattr(dp, column, value)
+            session.commit()
+        else:
+            raise ValueError(f"no database entries which meet filter criteria {query}")
```

Comment:
```suggestion
        if len(DataPackages) == 0:
            raise ValueError(f"no database entries which meet filter criteria {query}")
            return
        for dp in DataPackages:
            for column, value in column_value.items():
                setattr(dp, column, value)
        session.commit()  
```

Doesnt really apply here but to reduce cognitive complexity we can do the following.
Generally putting a loop inside an if statement raises complexity considerable when checked using static code analyzers.

Reply:
that makes perfect sense